src/main/java/net/dries007/tfc/objects/blocks/devices/BlockFirePit.java

```diff
@@ -273,6 +320,46 @@ else if (tile.getCookingPotStage() == TEFirePit.CookingPotStage.FINISHED)
             {
                 TFCGuiHandler.openGui(worldIn, pos, player, TFCGuiHandler.Type.FIRE_PIT);
             }
+            else if ((held == ItemStack.EMPTY) && (attachment != FirePitAttachment.NONE))
+            {
+                boolean anythingInTheInv = false;
+                IItemHandler cap = tile.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null);
+                for (int i = TEFirePit.SLOT_EXTRA_INPUT_START; i <= TEFirePit.SLOT_EXTRA_INPUT_END; i++)
+                {
+                    if(!cap.getStackInSlot(i).isEmpty())
+                    {
+                        anythingInTheInv = true;
+                        break;
+                    }
+                }
+                if (!anythingInTheInv)
+                {
+                    if (state.getValue(LIT))
+                    {
+                        if (attachment == FirePitAttachment.COOKING_POT)
+                        {
+                            player.attackEntityFrom(DamageSourcesTFC.SOUP, 1);
+                        }
+                        else
+                        {
+                            player.attackEntityFrom(DamageSourcesTFC.GRILL, 1);
+                        }
+
+                    }
+                    else
+                    {
+                        tile.onRemoveAttachment();
+                        worldIn.setBlockState(pos, state.withProperty(ATTACHMENT, FirePitAttachment.NONE));
+                        ItemStack attachmentItem = ItemStack.EMPTY;
+                        if (attachment == FirePitAttachment.COOKING_POT)
+                            attachmentItem = ItemsTFC.FIRED_POT.getDefaultInstance();
+                        if (attachment == FirePitAttachment.GRILL)
+                            attachmentItem = ItemsTFC.WROUGHT_IRON_GRILL.getDefaultInstance();
+                        player.addItemStackToInventory(attachmentItem);
```

Comment:
No, this needs to give the actual item instance back to the player that is held by the tile. That's *why it is held there*.

Reply:
Also, you CANNOT call `Item#getDefaultInstance` here as it is `@SideOnly(Side.CLIENT)`, which will crash and burn on dedicated servers.